src/weights.jl

```diff
@@ -2,44 +2,181 @@
+
 """
-    WeightVec(vs, [wsum])
+    bias(n::Integer, [corrected])
 
-Construct a `WeightVec` with weight values `vs` and sum of weights `wsum`.
+Computes the corrected (default) or uncorrected bias for any `n` observations.
+
+```math
+\fraction{1}{n - 1}
+```
+"""
+bias(n::Integer, corrected=true) = inv(n - Int(corrected))
+
+"""
+    bias(w::AbstractWeights, [corrected])
+
+Computes the corrected (default) or uncorrected bias for any weight vector.
+The default equation assumes analytic/precision/reliability weights and determines the
+bias as:
+
+```math
+\fraction{1}{∑w × (1 - ∑(w'²))}
+```
+where w' represents the normalized weights
+"""
+function bias(w::AbstractWeights, corrected=true)
+    s = sum(w)
+    if corrected
+        return inv(s * (1 - sum(normalize(values(w), 1) .^ 2)))
+    else
+        return inv(s)
+    end
+end
+
+@weights AnalyticWeights
+
+"""
+    AnalyticWeights(vs, [wsum])
+
+Construct a `AnalyticWeights` with weight values `vs` and sum of weights `wsum`.
 If omitted, `wsum` is computed.
 """
-function WeightVec{S<:Real, V<:RealVector}(vs::V, s::S=sum(vs))
-    return WeightVec{S, eltype(vs), V}(vs, s)
+function AnalyticWeights{S<:Real, V<:RealVector}(vs::V, s::S=sum(vs))
+    return AnalyticWeights{S, eltype(vs), V}(vs, s)
 end
 
 """
+    aweights(vs)
+
+Construct a `AnalyticWeights` type from a given array.
+"""
+aweights(vs::RealVector) = AnalyticWeights(vs)
+aweights(vs::RealArray) = AnalyticWeights(vec(vs))
+
+"""
     weights(vs)
 
-Construct a `WeightVec` from a given array.
+Alias for aweights(vs)
+"""
+weights(vs) = aweights(vs)
+
+@weights FrequencyWeights
+
+function FrequencyWeights{S<:Integer, V<:IntegerVector}(vs::V, s::S=sum(vs))
+    return FrequencyWeights{S, eltype(vs), V}(vs, s)
+end
+
+"""
+    fweights(vs)
+
+Construct a `FrequencyWeights` type from a given array.
+"""
+fweights(vs::IntegerVector) = FrequencyWeights(vs)
+fweights(vs::IntegerArray) = FrequencyWeights(vec(vs))
+
 """
-weights(vs::RealVector) = WeightVec(vs)
-weights(vs::RealArray) = WeightVec(vec(vs))
+    bias(w::FrequencyWeights, [corrected])
 
-eltype(wv::WeightVec) = eltype(wv.values)
-length(wv::WeightVec) = length(wv.values)
-values(wv::WeightVec) = wv.values
-sum(wv::WeightVec) = wv.sum
-isempty(wv::WeightVec) = isempty(wv.values)
+```math
+\fraction{1}{∑w - 1}
+```
+"""
+bias(w::FrequencyWeights, corrected=true) = inv(sum(w) - Int(corrected))
+
+@weights ProbabilityWeights
+
+function ProbabilityWeights{S<:Real, V<:RealVector}(vs::V, s::S=sum(vs))
+    return ProbabilityWeights{S, eltype(vs), V}(vs, s)
+end
 
-Base.getindex(wv::WeightVec, i) = getindex(wv.values, i)
-Base.size(wv::WeightVec) = size(wv.values)
+"""
+    pweights(vs)
+
+Construct a `ProbabilityWeights` type from a given array.
+"""
+pweights(vs::RealVector) = ProbabilityWeights(vs)
+pweights(vs::RealArray) = ProbabilityWeights(vec(vs))
 
+"""
+    bias(w::ProbabilityWeights, [corrected])
+
+```math
+\fraction{n}{∑w × (n - 1)}
```

Comment:
I'm fine with just using LaTeX commands, but the julia docs recommend using unicode characters.

>Use Unicode characters rather than their LaTeX escape sequence, i.e. ``α = 1`` rather than ``\\alpha = 1``.